fretboardgtr/fretboard.py

```diff
@@ -234,14 +243,111 @@ def add_note(self, string_no: int, note: str, root: Optional[str] = None) -> Non
+            root: str,
+            fingers: FingerSet
+    ) -> None:
+        if (fret is not None):
+            frets_on_string = self._get_frets_for_string(string_no, fingers)
+            # if "None" is part of frets on string, then the string is to be muted
+            if (None not in frets_on_string and fret == max(frets_on_string)):
+                self.add_single_note_from_index(string_no, fret, root)
+
+    def add_notes_from_fingerset(
+        self, fingers: FingerSet, root: Optional[str] = None
+    ) -> None:
+        for finger in fingers:
+            strings = finger[0]
+            fret = finger[1]
+            if (isinstance(strings,Tuple)):
+                for s in range(strings[0], strings[1]+1):
+                    self._add_note_for_fingerset(s, fret, root, fingers)
+            else:
+                self._add_note_for_fingerset(strings, fret, root, fingers)
+
+    def add_fingerset(self, fingers: FingerSet, show_text=True ) -> None:
+        for finger in fingers:
+            string_no = finger[0]
+            frets = finger[1]
+            if (len(finger) == 3) and show_text:
+                text=finger[2]
+            else:
+                text = ""
+
+            f : FingerPos = (string_no, frets, text)
+            self.add_fingerpos(f)
+
+    def add_fingerpos(
+        self, fingerpos : FingerPos
+    ) -> None:
+
+        if (isinstance(fingerpos[0], Tuple)):
+            self.add_finger_element(fingerpos[0], fingerpos[1], fingerpos[2])
+
+        elif (isinstance(fingerpos[0], int)) :
+            if (fingerpos[1] is None):
+                string_good = self.fretboard.get_string_with_good_index(fingerpos[0])
+                position = self.fretboard.get_cross_position(string_good)
+                cross = Cross(position, config=self.config.cross)
+                self.elements.crosses.append(cross)
+            else:
+                self.add_finger_element((fingerpos[0], fingerpos[0]), fingerpos[1], fingerpos[2])
+            pass
+
+    def _get_finger_element(
+        self, position_from: Tuple[float, float], position_to: Tuple[float, float], text: str) -> Finger:
+        config = copy.copy(self.config.fingers)
+
+        pos, size = self.fretboard.get_bounding_box(position_to, position_from, config.radius)
+        _finger = Finger(pos, size, text, config=config)
+        return _finger
+
+    def add_finger_element(self, strings: Tuple[int,int], fret: int, text: Optional[str] = "") -> None:
+        """Build and add barre element."""
+        if strings[0] < 0 or strings[0] > len(self.tuning):
+            raise ValueError(f"String 'from' number is invalid. Tuning is {self.tuning}")
+
+        if strings[1] < 0 or strings[1] > len(self.tuning):
+            raise ValueError(f"String 'to' number is invalid. Tuning is {self.tuning}")
+
+        index = fret - self.config.general.first_fret + 1
+        if (index < 1 ) or (index > (self.config.general.last_fret - self.config.general.first_fret)):
```

Comment:
maybe use `from_string` and `to_string` in the definition to be more explicit with the errors ?